recipes-bsp/u-boot/files/0022-Add-a-reserved-memory-for-watchdog.patch

```diff
@@ -0,0 +1,31 @@
+From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
+From: Li Hua Qian <huaqian.li@siemens.com>
+Date: Wed, 7 Jun 2023 11:29:59 +0800
+Subject: [PATCH] Add a reserved memory for watchdog
+
+To have the WDIOF_CARDRESET support for the TI AM65x platform watchdog,
+this patch reserves the memory, which indicates if the current boot due
+to a watchdog reset.
+
+Signed-off-by: Li Hua Qian <huaqian.li@siemens.com>
+---
+ arch/arm/dts/k3-am65-iot2050-common.dtsi | 5 +++++
+ 1 file changed, 5 insertions(+)
+
+diff --git a/arch/arm/dts/k3-am65-iot2050-common.dtsi b/arch/arm/dts/k3-am65-iot2050-common.dtsi
+index 65da2268..e208deba 100644
+--- a/arch/arm/dts/k3-am65-iot2050-common.dtsi
++++ b/arch/arm/dts/k3-am65-iot2050-common.dtsi
+@@ -64,6 +64,12 @@
+ 			alignment = <0x1000>;
+ 			no-map;
+ 		};
++
++		/* To reserve the power-on(PON) reason for watchdog reset */
++		wdt_reset_memory_region: wdt-memory@a2200000 {
++			reg = <0x00 0xa2200000 0x00 0x00001000>;
++			no-map;
++		};
```

Comment:
Still missing the second hunk of the kernel patch.

Reply:
Even if the second hunk is unnecessary,  do we have to add it in the uboot patch?

The second hunk is used by kernel driver to get the reserved address.

Reply:
The U-Boot dts files are by now (in upstream) copies of the kernel files. So, yes, we want this, even if things won't be sync'ed anymore for old 2022.10, the patch may still apply to upcoming 2023.10.